conda/testing/solver_helpers.py

```diff
@@ -24,6 +24,42 @@
 from . import helpers
 
 
+@pytest.fixture(params=["libmamba", "classic"], autouse=True)
+def parametrized_solver_fixture(request, monkeypatch):
+    """
+    A parameterized fixture that sets the solver backend to (1) libmamba
+    and (2) classic for each test. It's using autouse=True, so only import it in
+    modules that actually need it.
+
+    Note that skips and xfails need to be done _inside_ the test body.
+    Decorators can't be used because they are evaluated before the
+    fixture has done its work!
+
+    So, instead of:
+
+        @pytest.mark.skipif(context.solver == "libmamba", reason="...")
+        def test_foo():
+            ...
+
+    Do:
+
+        def test_foo():
+            if context.solver == "libmamba":
+                pytest.skip("...")
+            ...
+    """
```

Comment:
I dislike that we have to rely on the implicitness of imports to trigger such a drastic change to the test suite since it breaks the contract of dependency injection promised by pytest.

The pytest canonical way instead is using [`usefixtures`](https://docs.pytest.org/en/stable/how-to/fixtures.html#use-fixtures-in-classes-and-modules-with-usefixtures) on the module level and NOT use autouse for it, e.g.:

```
pytestmark = pytest.mark.usefixtures("parametrized_solver_fixture")
```

Reply:
Oh nice, didn't know about this method. Definitely better. Thanks!